Adding `spin-registry` support (#346)
fermyon/developer: content/spin/spin-oci.md

$ spin --version
```

> Please note if you're using version v0.9.0 or later you will be using the command `spin registry`. The command `spin oci` will only be supported with v0.8.8. 

Reviewer: Looks like https://github.com/fermyon/developer/commit/6e163fbabea14119032c7a7455bd1882bda2393e already covered this off.

> Please note if you're using version v0.9.0 or later you will be using the command `spin registry`. The command `spin oci` will only be supported with v0.8.8. 

Reviewer: ```suggestion
If you're using version v0.9.0 or later, it is officially recommended that you use the command `spin registry`. However, please note that the command `spin oci` (whilst initially supported by v0.8.0) is now also an alias (`oci` and `registry` are interchangeable) in v0.9.0. 
```

> Please note if you're using version v0.9.0 or later you will be using the command `spin registry`. The command `spin oci` will only be supported with v0.8.8. 

Reviewer: Thanks @radu-matei ; all fixed.

```

Now we're ready to push the application. Run the `spin oci push` command to push your application to the registry. 

{{ tabs "spin-version" }}

{{ startTab "v0.8.0"}}

 <!-- @selectiveCpy -->

 ```bash
 spin oci push ghcr.io/USERNAME/spin-react-fullstack:v1
```

{{ blockEnd }}

{{ startTab "v0.9.0"}}

 <!-- @selectiveCpy -->

 ```bash
 spin registry push ghcr.io/USERNAME/spin-react-fullstack:v1
```

{{ blockEnd }}

{{ blockEnd }}

You now have a Spin application stored in your registry. You can see the artifact under packages in the [GitHub UI](https://docs.github.com/en/packages/learn-github-packages/viewing-packages#viewing-a-repositorys-packages).

## Pull a Spin App From GHCR

Now that we've successfully pushed a Spin app, let's see if we can pull it. To do so, run the following command: 

{{ tabs "spin-version" }}

{{ startTab "v0.8.0"}}

 <!-- @selectiveCpy -->
 
 ```bash
 spin oci pull ghcr.io/USERNAME/spin-react-fullstack
```

{{ blockEnd }}

{{ startTab "v0.9.0"}}

 <!-- @selectiveCpy -->
 
 ```bash

>  spin registry pull ghcr.io/USERNAME/spin-react-fullstack

Reviewer 2: This also needs the tag.

Reviewer: Thanks @radu-matei

>  spin registry pull ghcr.io/USERNAME/spin-react-fullstack

Reviewer: ```suggestion
 spin registry pull ghcr.io/USERNAME/spin-react-fullstack:v1
```

```

{{ blockEnd }}

{{ blockEnd }}

## Run a Spin App From GHCR

Lastly, let's run this Spin application. 

{{ tabs "spin-version" }}

{{ startTab "v0.8.0"}}

 <!-- @selectiveCpy -->

 ```bash
 # make sure you've built your application with `spin build` prior

>  spin oci run ghcr.io/USERNAME/spin-react-fullstack

Reviewer 2: This also needs the tag.

Reviewer: Thanks @radu-matei

>  spin oci run ghcr.io/USERNAME/spin-react-fullstack

Reviewer: ```suggestion
 spin up --from-registry ghcr.io/USERNAME/spin-react-fullstack:v1
```

```

{{ blockEnd }}

{{ startTab "v0.9.0"}}

<!-- @selectiveCpy -->

 ```bash
 # make sure you've built your application with `spin build` prior

>  spin up --from-registry ghcr.io/USERNAME/spin-react-fullstack

Reviewer 2: This also needs the tag.

Reviewer: Thanks @radu-matei

>  spin up --from-registry ghcr.io/USERNAME/spin-react-fullstack

Reviewer: ```suggestion
 spin up --from-registry ghcr.io/USERNAME/spin-react-fullstack:v1
```
